machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,348 @@
+
+	// paused specifies whether or not changes to this machine config pool should be stopped.
+	// This includes generating new desiredMachineConfig and update of machines.
+	Paused bool `json:"paused" protobuf:"varint,3,opt,name=paused"`
+
+	// maxUnavailable specifies the percentage or constant number of machines that can be updating at any given time.
+	// default is 1.
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty" protobuf:"bytes,4,opt,name=maxUnavailable"`
+
+	// The targeted MachineConfig object for the machine config pool.
+	Configuration MachineConfigPoolStatusConfiguration `json:"configuration" protobuf:"bytes,5,opt,name=configuration"`
+}
+
+// MachineConfigPoolStatus is the status for MachineConfigPool resource.
+type MachineConfigPoolStatus struct {
+	// observedGeneration represents the generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
+
+	// configuration represents the current MachineConfig object for the machine config pool.
+	Configuration MachineConfigPoolStatusConfiguration `json:"configuration" protobuf:"bytes,2,opt,name=configuration"`
+
+	// machineCount represents the total number of machines in the machine config pool.
+	MachineCount int32 `json:"machineCount" protobuf:"varint,3,opt,name=machineCount"`
+
+	// updatedMachineCount represents the total number of machines targeted by the pool that have the CurrentMachineConfig as their config.
+	UpdatedMachineCount int32 `json:"updatedMachineCount" protobuf:"varint,4,opt,name=updatedMachineCount"`
+
+	// readyMachineCount represents the total number of ready machines targeted by the pool.
+	ReadyMachineCount int32 `json:"readyMachineCount" protobuf:"varint,5,opt,name=readyMachineCount"`
+
+	// unavailableMachineCount represents the total number of unavailable (non-ready) machines targeted by the pool.
+	// A node is marked unavailable if it is in updating state or NodeReady condition is false.
+	UnavailableMachineCount int32 `json:"unavailableMachineCount" protobuf:"varint,6,opt,name=unavailableMachineCount"`
+
+	// degradedMachineCount represents the total number of machines marked degraded (or unreconcilable).
+	// A node is marked degraded if applying a configuration failed..
+	DegradedMachineCount int32 `json:"degradedMachineCount" protobuf:"varint,7,opt,name=degradedMachineCount"`
+
+	// conditions represents the latest available observations of current state.
+	// +optional
+	Conditions []MachineConfigPoolCondition `json:"conditions" protobuf:"bytes,8,rep,name=conditions"`
+}
+
+// MachineConfigPoolStatusConfiguration stores the current configuration for the pool, and
+// optionally also stores the list of MachineConfig objects used to generate the configuration.
+type MachineConfigPoolStatusConfiguration struct {
+	corev1.ObjectReference `json:",inline" protobuf:"bytes,1,opt,name=objectReference"`
```

Comment:
You probably don't honor all the fields of this struct.  We encourage creating specific reference objects so you can provide precise docs.  This will need revisiting.